app/models/deterministic_assignment_creation.rb

```diff
@@ -37,8 +33,8 @@ def split
     @split ||= Split.find_by!(name: split_name)
   end
 
-  def assignment
-    @assignment ||= Assignment.find_or_initialize_by visitor: visitor, split: split
+  def existing_assignment
+    @existing_assignment ||= Assignment.find_by visitor: visitor, split: split
```

Comment:
We might want to drop the memoization from here. I _think_ we only call `existing_assignment` once now, and since we can't actually memoize the `nil` I wonder if this is just going to cause slight dev confusion going forward

Reply:
sure
```suggestion
    Assignment.find_by visitor: visitor, split: split
```